### PR TITLE
Add resources to profiler

### PR DESCRIPTION
**Pull request by piotmag769**
Closes https://github.com/foundry-rs/starknet-foundry/issues/1697

### CODE REVIEW

File: profiler/src/profile_builder.rs [hunk 1]
```diff
@@ -128,33 +139,47 @@ impl ProfilerContext {
...
+        },
+        pprof::ValueType {
+            r#type: context.string_id(&String::from("n_steps")).into(),
+            unit: context.string_id(&String::from("count")).into(),
+        },
+        pprof::ValueType {
+            r#type: context.string_id(&String::from("n_memory_holes")).into(),
+            unit: context.string_id(&String::from("count")).into(),
+        },
+    ];
```
Comment (piotmag769) [inline profiler/src/profile_builder.rs:161]: Basic measurement types, will be always there

File: profiler/src/trace_reader.rs [hunk 2]
```diff
@@ -77,26 +122,73 @@ pub fn collect_samples_from_trace(trace: &CallTrace) -> Vec<Sample> {
...
+                .flat_resources
+                .vm_resources
+                .builtin_instance_counter
+                .keys()
+                .cloned(),
+        );
+        syscalls.extend(sample.flat_resources.syscall_counter.keys())
+    }
+    ResourcesKeys { syscalls, builtins }
+}
```
Comment (piotmag769) [inline profiler/src/trace_reader.rs:166]: This doesn't really require iteration rn (would be enough to get the keys from the top call), but since blockifier plans to change resources to not include inner calls we will prob change the semantics of the expected input file as well - then this iteration will be needed.
 
TL;DR: not needed, but more flexibile, doesn't do any harm

File: profiler/src/trace_reader.rs [hunk 3]
```diff
@@ -25,6 +28,48 @@ pub enum SampleType {
...
+    pub flat_resources: ExecutionResources,
+}
+
+impl Sample {
+    pub fn extract_measurements(
+        &self,
+        measurement_types: &[ValueType],
+        context: &ProfilerContext,
+    ) -> Vec<i64> {
+        let mut measurements_map: HashMap<&str, i64> = vec![
```
Comment (piotmag769) [inline profiler/src/trace_reader.rs:40]: This is prob not the best way to do it but couldn't think of anything better. For sure it's better than bunch of brute force ifs or coupling parts of code that shouldn't be coupled together (keep in mind measurement values order is dependent on measurement types order)

File: profiler/src/main.rs [hunk 4]
```diff
@@ -4,6 +4,7 @@ use std::{
     path::Path,
 };
 
+use crate::trace_reader::collect_resources_keys;
```
Comment (war-in) [inline profiler/src/main.rs:7]: this import should be merged with the one below (which I cannot point here)

```use trace_reader::collect_samples_from_trace;```

File: profiler/src/profile_builder.rs [hunk 5]
```diff
@@ -47,18 +48,28 @@ impl ProfilerContext {
     fn new() -> Self {
         ProfilerContext {
             strings: vec![("".to_string(), StringId(0))].into_iter().collect(),
+            id_to_strings: vec![(StringId(0), "".to_string())].into_iter().collect(),
```
Comment (war-in) [inline profiler/src/profile_builder.rs:51]: Maybe `id_to_string`? the name suggests that one id points to multiple strings

**Review by war-in — COMMENTED** [review]
Would be nice to get a screenshot of the generated diagram

**Review by war-in — APPROVED** [review]

**Review by war-in — APPROVED** [review]
